sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -114,14 +114,16 @@ def test_hash_empty_input():
 
 
 def test_hasher_invalid_input():
+    raw_X = [[], (), iter(range(0))]
+
     with pytest.raises(ValueError):
-        FeatureHasher(input_type="gobbledygook")
+        FeatureHasher(input_type="gobbledygook").transform(raw_X)
```

Comment:
To make sure that `FeatureHasher` raises during `transform` and not `__init__`:

```python
    h = FeatureHasher(input_type="gobbledygook")
    with pytest.raises(ValueError):
        h.transform(raw_X)
```

Same comment applies to the changes below.

Reply:
Thank you for your comment. We have changed the test accordingly.